src/persistent_backend_manager.cc

```diff
@@ -0,0 +1,113 @@
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#include "persistent_backend_manager.h"
+
+#include <memory>
+#include <mutex>
+#include "backend_config.h"
+#include "filesystem/api.h"
+#include "triton/common/logging.h"
+
+namespace triton { namespace core {
+
+static std::weak_ptr<PersistentBackendManager> persist_backend_manager_;
+static std::mutex mu_;
+
+Status
+PersistentBackendManager::Create(
+    const triton::common::BackendCmdlineConfigMap& config_map,
+    std::shared_ptr<PersistentBackendManager>* manager,
+    std::shared_ptr<TritonBackendManager>& backend_manager)
+{
+  std::lock_guard<std::mutex> lock(mu_);
+
+  // If there is already a manager then we just use it...
+  *manager = persist_backend_manager_.lock();
+  if (*manager != nullptr) {
+    return Status::Success;
+  }
+
+  manager->reset(new PersistentBackendManager());
+  persist_backend_manager_ = *manager;
+
+  RETURN_IF_ERROR(
+      (*manager)->InitPersistentBackends(config_map, backend_manager));
+  return Status::Success;
+}
+
+Status
+PersistentBackendManager::InitPersistentBackends(
+    const triton::common::BackendCmdlineConfigMap& config_map,
+    std::shared_ptr<TritonBackendManager>& backend_manager)
+{
+  for (const auto& be : {"pytorch"}) {
+    RETURN_IF_ERROR(InitPersistentBackend(be, config_map, backend_manager));
+  }
+
+  return Status::Success;
+}
+
+Status
+PersistentBackendManager::InitPersistentBackend(
+    const std::string& backend_name,
+    const triton::common::BackendCmdlineConfigMap& config_map,
+    std::shared_ptr<TritonBackendManager>& backend_manager)
+{
+  std::string backends_dir;
+  std::string specialized_backend_name;
+  std::string backend_libname;
+  RETURN_IF_ERROR(
+      BackendConfigurationGlobalBackendsDirectory(config_map, &backends_dir));
+  RETURN_IF_ERROR(BackendConfigurationSpecializeBackendName(
+      config_map, backend_name, &specialized_backend_name));
+  RETURN_IF_ERROR(BackendConfigurationBackendLibraryName(
+      specialized_backend_name, &backend_libname));
+
+  const auto backend_dir = JoinPath({backends_dir, specialized_backend_name});
+  const auto backend_libpath = JoinPath({backend_dir, backend_libname});
+  bool exists = false;
+  RETURN_IF_ERROR(FileExists(backend_libpath, &exists));
+  if (exists) {
+    triton::common::BackendCmdlineConfig empty_backend_cmdline_config;
+    const triton::common::BackendCmdlineConfig* config;
+    const auto& itr = config_map.find(backend_name);
+    if (itr == config_map.end()) {
+      config = &empty_backend_cmdline_config;
+    } else {
+      config = &itr->second;
+    }
+
+    std::shared_ptr<TritonBackend> persist_backend;
+    RETURN_IF_ERROR(backend_manager->CreateBackend(
+        backend_name, backend_dir, backend_libpath, *config, &persist_backend));
+    persist_backends_.push_back(persist_backend);
```

Comment:
I don't think you need a dedicated class for this, basically you just want the core to hold its own reference to the backend to keep its reference count > 0. The `list<shared_ptr<Backend>>` can be hold within backend manager?

Reply:
Fixed.